exercises/operators/solution/operators_sol.cpp

```diff
@@ -0,0 +1,135 @@
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <compare>
+#include <numeric>
+
+class Fraction {
+
+public:
+
+  Fraction(int a_num, int a_denom = 1) : m_num(a_num), m_denom(a_denom) {}
+
+  std::string str() const {
+    std::ostringstream oss;
+    oss << m_num << '/' << m_denom;
+    return oss.str();
+  }
+
+  friend bool operator==(Fraction const & lhs, Fraction const & rhs) {
+    return lhs.m_num == rhs.m_num && lhs.m_denom == rhs.m_denom;
+  }
+
+  friend bool operator!=(Fraction const & lhs, Fraction const & rhs) {
+    return !(lhs==rhs);
+  }
+
+  friend auto operator<=>( Fraction const & lhs, Fraction const & rhs ) {
+    return ((lhs.m_num*rhs.m_denom)<=>(rhs.m_num*lhs.m_denom));
+  }
```

Comment:
You do not need operaot== and operator!= anymore I believe (plese double check, I'm not yet at ease with this and I did not read back the slides)

Reply:
Actually no. I beleive that generally one must implement operator== reusing <=>. Yet, since we use a Fraction, and provided we do not normalize() systematically, we are exactly in the use case where it makes sense to emphasize the difference between equality (operator==) and equivalence (operator<=>()==0). In the solution of the exercise, operator== check that both numerators and denomiators are exactly the same, although <=> only compare a.num*b.denom with a.denom*b.num.

Reply:
Indeed you're 100% right. I should probably follow this course myself :rofl: